app/views/flickr_items/_flickr_item.html.erb

```diff
@@ -1,5 +1,8 @@
 <h2><%= flickr_item.title %></h2>
 
-<a data-flickr-embed="true" href="https://www.flickr.com/photos/46626331@N07/<%= flickr_item.photo_id %>"><img src="https://farm9.staticflickr.com/8707/<%= flickr_item.photo_id %>_5908ab7d1d_k.jpg" alt=<%= flickr_item.title %>></a>
+<a data-flickr-embed="true" href="<%= flickr_item.flickr_url(FlickrDetail.find_by_user_id(flickr_item.blog_item.user_id).nsid, flickr_item.photo_id) %>">
```

Comment:
You need some method for that one: "FlickrDetail.find_by_user_id(flickr_item.blog_item.user_id).nsid"

Maybe define it as flickr_item#owner_nsid ?